docs/SupportedONNXOps-NNPA.md

```diff
@@ -12,32 +12,32 @@ Onnx-mlir currently supports ONNX operations targeting up to opset 18. Limitatio
-| **GlobalAveragePool** |1 |- Input shape must be 4D tensor(NCHW).<br>- Dimensions in `H` and `W` must be static. | |
-| **LSTM** |14 |- `direction` and `hidden_size` in `W` must have static dimensions.<br>- `R` must have static dimensions.<br>- `B` and `initial_h` have static dimensions if given. `B`'s direction dim must be 1 or 2.<br>- `P`(peepholes), `activation_alpha`, and `activation_beta` are not supported.<br>- `activations` must be `["Sigmoid", "Tanh", "Tanh"]`.<br>- `clip` is not supported.<br>- `input_forget` must be default value(0).<br>- `layout` is not supported. | |
-| **LeakyRelu** |16 |The operations immediately before and after the LeakyRelu operation must be executed on the NNPA. Otherwise, LeakyRelu is executed on the CPU. This limitation is set to avoid performance degradation. | |
-| **Log** |13 |Input tensor must have 4 dimensions. | |
-| **LogSoftmax** |13 | | |
-| **MatMul** |13 |Ranks of input tensors must be (Rank of A, Rank of B) = (M, N), where M >= 2 and N >= 2. | |
-| **Max** |13 |- Shape of input tensors must be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
-| **MaxPool** |12 |- `auto_pad` must be `NOTSET`, `VALID`, and `SAME_UPPER`. If `NOTSET` is used, `pads` must be set so that the padding valid type or same upper.<br>- `ceil_mode` must be default value(0) <br>- Input and output tensors must be 4D tensors(N x C x H x W).<br>- `kernel_shape` must be static.<br>- `ceil_mode` must be default value(0).<br>- `dilations` must be default value(1). | |
-| **Min** |13 |- Shape of input tensors must be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
-| **Mul** |14 |- Shape of input tensors should be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
-| **Pow** |15 |- Exponent should be a scalar integer and less or equal to 64. | |
-| **ReduceMean** |18, 13 |- `keepdims` must be 1.<br>- Input tensor must be 4D tensors and `axis` must be [2, 3]. | |
-| **Relu** |14 |Input tensor must be less than or equal to 4 dimensions. | |
-| **Sigmoid** |13 |Input tensor must be less than or equal to 4 dimensions. | |
-| **Softmax** |13, 11 |- `axis` must be the last dimension, i.e. `rank - 1` or -1. | |
-| **Sub** |14 |- Shape of input tensors should be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
-| **Sum** |13 |- All inputs must have the same static shape (Broadcasting not supported.)<br>- Single input not supported. | |
-| **Tanh** |13 |Input tensor must be less than or equal to 4 dimensions. | |
+| Op |Min Opset |Max Opset |Limitations |Notes |
+| --- |--- |--- |--- |--- |
+| **Add** |14 |18 |- Shape of input tensors must be the same since broadcasting is not supported.<br>- Input tensors must have static dimensions. | |
```

Comment:
My intent is not to be aggressive when specifying the min but rather to err on the side of caution. In my mind I would rather be safe than sorry. However, I tested `inception-v2-6.onnx` myself and it does compile. So we can still support Add.6 but not the attributes. I can do another pass at the documentation and make updates to every other operation as well. Thanks

Reply:
I also tested using `bvlcalexnet-3.onnx`, so it appears we do not support anything older than opset 6. So I can update the document to use Add.6, Conv.11 and BatchNormalization.6 :

```
meganhampton@megans-mbp-2 build % ./Debug/bin/onnx-mlir --O3 --EmitLib bvlcalexnet-3.onnx --onnx-op-stats TXT -v 
Warning: ONNX Conv in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Relu in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX LRN in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX MaxPool in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Conv in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Relu in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX LRN in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX MaxPool in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Conv in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Relu in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Conv in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Relu in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Conv in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Relu in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX MaxPool in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Gemm in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Relu in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Dropout in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Gemm in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Relu in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Dropout in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Gemm in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
Warning: ONNX Softmax in your model is using Opset 3, which is quite old. Please consider regenerating your model with a newer Opset.
```

Reply:
I believe opset 3 and below are antique... the return on investment to support these is very very low. We should ignore these cases.

Reply:
Agreed. Anything less than 6 can be ignored. Six is already pretty old so we wouldn't target them either but they all already work. There was some talk of getting the zoo itself deprecate/remove models other than 6 a while back (year ago?) but I haven't heard much about it since then.